app/src/main/java/org/mozilla/focus/searchsuggestions/SearchSuggestionsFetcher.kt

```diff
@@ -73,6 +76,22 @@ class SearchSuggestionsFetcher(searchEngine: SearchEngine) : CoroutineScope {
         }
     }
 
+    private fun checkSearchEngineShortcut(query: String): List<String>? {
+        val searchEngineShortcutAndQuery = UrlUtils.splitShortcutFromQuery(query)
+
+        if (searchEngineShortcutAndQuery != null) {
+            val searchEngineShortcut = searchEngineShortcutAndQuery[0]
+            val matchingSearchEngines = UrlUtils.searchEngineShortcutsMap.keys.toList()
```

Comment:
A few observations
1. string will never be null since our query string is never null - all we have to do is check if it's empty
2. we don't need to filter since we should only be matching one key at a time

which means we can replace code with this
```
private fun checkSearchEngineShortcut(query: String): List<String>? {
        val searchEngineShortcutAndQuery = UrlUtils.splitShortcutFromQuery(query)
        if (searchEngineShortcutAndQuery.isEmpty()) return null

        val searchEngineShortcut = searchEngineShortcutAndQuery[0]
        return if (UrlUtils.searchEngineShortcutsMap.containsKey(searchEngineShortcut)) {
            listOf(searchEngineShortcut)
        } else {
            null
        }
    }
```

Reply:
Regarding replacing the Fitler with contains ... This function is called repeatedly as user types in the search bar and returns the appropriate suggestions for search shortcuts. I was trying to emulate how the search shortcut works on firefox desktop, where as soon as you type `@`, all the possible shortcuts appear. As you continue typing it tries to filter out the shortcuts based on what your input. If we were to replace filter with contains, we would not get the same sort of behaviour that we see on Firefox desktop.

Reply:
Ahhh ok cool then ignore it haha. It's perfect!